src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final String INDEX_EXCEPTION_TEXT = "Index is out of bounds!";
+    private static final String NO_SUCH_ELEMENT_EXCEPTION = "Element not found in list!";
+
+    private T[] items = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (size == items.length) {
+            items = grow();
```

Comment:
```suggestion
            items = grow();
```

lets move `if (size == items.length) {` check to grow() method